ANMN/burst_averaged_product/burst_average.py

```diff
@@ -159,8 +159,10 @@ def burst_average_data(time_values, var_values, var_qc_exclusion):
     # look for start index of each burst, ie the index of each new spike of
     # difft variable
     idx_spike = [0] # initialise first index
-    spikes    = np.where(difft > 1)[0] + 1
+    spikes    = np.where(difft > INSTRUMENT_SAMPLE_INTERVAL)[0] + 1
     idx_spike.extend(spikes)
+    if idx_spike == [0]:
+        raise Exception('No burst was found in time values - FV01 is probably already averaged')
```

Comment:
A neater way to say this is:
```python
assert idx_spike == [0], 'No burst was found in time values'
```

Reply:
If you want to use an assertion a you should write 

    assert idx_spike != [0], 'No burst was found in time values'

since the assert function check that the statement is true and throw the error if not

Reply:
good spot !

Reply:
You shouldn't use assert for error handling in place of exceptions, they perform a different task.

https://wiki.python.org/moin/UsingAssertionsEffectively

```Assertions should *not* be used to test for failure cases that can occur because of bad user input or operating system/environment failures, such as a file not being found. Instead, you should raise an exception, or print an error message, or whatever is appropriate. One important reason why assertions should only be used for self-tests of the program is that assertions can be disabled at compile time. ```

If the user supplied input file is ultimately the source of the error, then it counts as "bad user input".

So if you're not happy for the assert statement to be stripped out of the code and have it still *fully functional*, it should be an explicit if statement and standard exception.

Reply:
Thanks for clarifying that @lwgordonimos

Reply:
Ah, good point. Thanks.